File: ts-knowledge-base/scripts/query.js
const fs = require('fs');
const https = require('https');

function post(host, path, body, headers) {
    return new Promise((resolve, reject) => {
        const options = { hostname: host, path: path, method: 'POST',
            headers: Object.assign({'Content-Type':'application/json','Accept':'application/json, text/event-stream'}, headers||{}) };
        const req = https.request(options, res => {
            let raw = '';
            res.on('data', chunk => { raw += chunk.toString(); });
            res.on('end', () => resolve(raw));
        });
        req.on('error', reject);
        req.write(JSON.stringify(body));
        req.end();
    });
}

async function query(question, isDeepSearch, userId) {
    const raw = await post('api-ai.h3c.com', '/deepsearch/api-search/v1/streamv1', {
        task: question,
        source: 'www,pmo,cjg,cmp,press,wwwen,ts_sqcs_knowleged',
        user_id: userId || 'z62875',
        session_id: require('crypto').randomUUID(),
        application: 'zhidao',
        searchonly: false,
        outline: false
    }, {});
    
    const lines = raw.split('\n').filter(l => l.startsWith('data: '));
    let finalAnswer = '';
    let references = [];
    
    for (const line of lines) {
        try {
            const obj = JSON.parse(line.substring(6));
            if (obj.code !== 200) continue;
            const meta = obj.data && obj.data.metadata;
            if (!meta) continue;
            const title = meta.title || '';
            const content = obj.data.content || '';
            
            if (title === 'Final Answer' || title === 'Outline Answer') {
                finalAnswer += content;
            }
            if (title === 'Reference information' && obj.data.content_type === 'list') {
                try { references = JSON.parse(content); } catch(e) {}
            }
        } catch(e) {}
    }
    
    return { answer: finalAnswer, references: references };
}

async function main() {
    const question = process.argv[2] || 'R4900 G7 服务器参数';
    const isDeep = process.argv.includes('--deep');
    const userId = 'z62875';
    
    console.log('问题: ' + question);
    console.log('深度思考: ' + isDeep);
    console.log('---------------');
    
    const result = await query(question, isDeep, userId);
    console.log(result.answer);
    
    if (result.references.length > 0) {
        console.log('\n--- 参考资料 ---');
        for (const ref of result.references.slice(0, 5)) {
            console.log('- ' + (ref.title || ref.name || ''));
            console.log('  ' + (ref.original_file_path || ref.url || ''));
        }
    }
}

main().catch(e => { console.error('Error:', e.message); process.exit(1); });

File: ts-knowledge-base/scripts/query.py
import sys
import json
import uuid
import http.client
import ssl

# ========== 配置 ==========
ACCOUNT = "ts_sqcs_knowleged"
PASSWORD = "TS_sqcs_knowleged_123"
SOURCES = ["www", "pmo", "cjg", "cmp", "press", "wwwen", "ts_sqcs_knowleged"]
BASE_URL = "https://api-searchservice.h3c.com"
TEST_URL = "https://api-searchservice-ts.h3c.com"
USE_TEST = True   # True=测试环境，False=生产环境


def get_host_port():
    url = TEST_URL if USE_TEST else BASE_URL
    host = url.replace("https://", "").split("/")[0]
    return host, 443


def build_payload(task, is_deep_search=False, user_id=None, session_id=None, source=None):
    return json.dumps({
        "isDeepSearch": is_deep_search,
        "task": task,
        "source": ",".join(source or SOURCES),
        "user_id": user_id or ACCOUNT,
        "session_id": session_id or str(uuid.uuid4()),
        "application": "zhidao",
        "searchonly": False
    }, ensure_ascii=False)


def query(question, is_deep_search=False, verbose=False):
    host, port = get_host_port()
    path = /itsearchserve/v1.0/search/getAiResultCommon
    payload = build_payload(question, is_deep_search=is_deep_search).encode("utf-8")

    ctx = ssl.create_default_context()
    conn = http.client.HTTPSConnection(host, port, timeout=60, context=ctx)
    try:
        conn.request("POST", path, body=payload, headers={
            "Content-Type": "application/json",
            "Accept": "application/json"
        })
        resp = conn.getresponse()
        raw = resp.read().decode("utf-8")
    except Exception as e:
        return f"[Error] 连接失败: {e}", []
    finally:
        conn.close()

    try:
        result = json.loads(raw)
    except Exception:
        return f"[Error] 解析响应失败\n原始内容: {raw[:500]}", []

    code = result.get("code", -1)
    if code != 0:
        return f"[Error] code={code}, msg={result.get("msg","")}", []

    data_obj = result.get("data", {})
    answer = data_obj.get("answer", "")
    references = data_obj.get("reference_information", []) or []

    if verbose and references:
        ref_lines = []
        for ref in references:
            title = ref.get("title", "")
            url = ref.get("original_file_path", "")
            ref_lines.append(f"- **{title}** + ("  \n  " + url if url else ""))
        if ref_lines:
            answer += "\n\n---\n**参考资料：**\n" + "\n".join(ref_lines)

    return answer, references


if __name__ == "__main__":
    if len(sys.argv) < 2:
        print("用法: python query.py <问题> [--deep]")
        sys.exit(1)
    question = sys.argv[1]
    is_deep = "--deep" in sys.argv
    print(f"问题: {question}")
    print(f"深度思考: {is_deep}")
    print(f"URL: {TEST_URL if USE_TEST else BASE_URL}")
    print("---------------")
    answer, refs = query(question, is_deep_search=is_deep, verbose=True)
    print(answer)